### ISSUE TITLE
Is GrainService support StorageProvider attribute?

### ISSUE THREAD

**Issue by SmartKarren** [dotnet/orleans#5363]
i tried to use GrainServices as documentation code.
StorageProvider  only described used for Grain class, can i use it like this?
and if StorageProvider not support ,how supposed i use storage in GrainService? thans

```
 [Reentrant]
 [StorageProvider(ProviderName = "PubSubStore")]
 public class LightstreamerDataService : GrainService, IDataService {
    readonly IGrainFactory GrainFactory;

    public LightstreamerDataService(IServiceProvider services, IGrainIdentity id, Silo silo, ILoggerFactory
         loggerFactory, IGrainFactory grainFactory) : base(id, silo, loggerFactory) {
        GrainFactory = grainFactory;
    }
    public Task MyMethod() {
    }
 }
```

**Comment by sergeybykov** [comment]
I don't think we ever considered or tried using declarative persistence for grain services. If it works, it would be by accident. In general, code inside a grain service class (or grain class) is free to use any persistence technology directly (without going via a persistence provider), so long as all IO calls are asynchronous.

**Comment by SmartKarren** [comment]
thanks. 
i read all documentation, test all sample codes,  but still fill confused to use orleans features **streams** and **observers** even after serach in github.
any advises or provide more examples to help us understand streams or observers?

**Comment by sergeybykov** [comment]
Could you list your questions and confusions. That would help us understand what's not clear.

**Comment by SmartKarren** [comment]
> Could you list your questions and confusions. That would help us understand what's not clear.

```
Streams Programming API documentation.

Current Status of Rewindable Streams: Both SMS and Azure Queue providers are not-rewindable and Orleans currently does not include an implementation of rewindable streams. We are actively working on this.
```
which stream providers support rewindable streams now?  and if i want impl my own providers, what should i do, is there any examples? thanks

**Comment by sergeybykov** [comment]
I submitter #5382 to update the outdated text.

**Comment by sergeybykov** [comment]
The doc issue got resolved via #5382.
